sdk/api_fingerprint.go

```diff
@@ -11,328 +11,118 @@ package sdk
-		// If we succeed, return the data, otherwise pass on to decode error.
-		err = a.client.decode(&localVarReturnValue, localVarBody, localVarHttpResponse.Header.Get("Content-Type"))
-
-		ParseResponse(localVarHttpResponse, &err)
-
-		return localVarReturnValue, localVarHttpResponse, err
-	}
-
-	if localVarHttpResponse.StatusCode >= 300 {
-		newErr := GenericSwaggerError{
-			body:  localVarBody,
-			error: localVarHttpResponse.Status,
-		}
-		if localVarHttpResponse.StatusCode == 200 {
-			var v Response
-			err = a.client.decode(&v, localVarBody, localVarHttpResponse.Header.Get("Content-Type"))
-			if err != nil {
-				newErr.error = err.Error()
-
-				ParseResponse(localVarHttpResponse, &newErr)
-
-				return localVarReturnValue, localVarHttpResponse, newErr
-			}
-			newErr.model = v
-
-			ParseResponse(localVarHttpResponse, &newErr)
-
-			return localVarReturnValue, localVarHttpResponse, newErr
-		}
-		if localVarHttpResponse.StatusCode == 403 {
-			var v ErrorVisits403
-			err = a.client.decode(&v, localVarBody, localVarHttpResponse.Header.Get("Content-Type"))
-			if err != nil {
-				newErr.error = err.Error()
+func createGetEventDefinition() requestDefinition {
+	return requestDefinition{
+		Path: func(args ...string) string {
+			pathParams := []string{"request_id"}
 
-				ParseResponse(localVarHttpResponse, &newErr)
+			path := "/events/{request_id}"
 
-				return localVarReturnValue, localVarHttpResponse, newErr
+			for i, arg := range args {
+				path = strings.Replace(path, "{"+pathParams[i]+"}", arg, -1)
 			}
-			newErr.model = v
 
-			ParseResponse(localVarHttpResponse, &newErr)
+			return path
+		},
+		StatusCodeResultsFactoryMap: map[int]func() interface{}{
+			200: func() interface{} { return &EventResponse{} },
+			403: func() interface{} { return &ErrorCommon403Response{} },
+			404: func() interface{} { return &ErrorEvent404Response{} },
+		},
+	}
+}
 
-			return localVarReturnValue, localVarHttpResponse, newErr
-		}
-		if localVarHttpResponse.StatusCode == 429 {
-			var v ManyRequestsResponse
-			err = a.client.decode(&v, localVarBody, localVarHttpResponse.Header.Get("Content-Type"))
-			if err != nil {
-				newErr.error = err.Error()
+func createGetVisitsDefinition() requestDefinition {
+	return requestDefinition{
+		Path: func(args ...string) string {
+			pathParams := []string{"visitor_id"}
 
-				ParseResponse(localVarHttpResponse, &newErr)
+			path := "/visitors/{visitor_id}"
 
-				return localVarReturnValue, localVarHttpResponse, newErr
+			for i, arg := range args {
+				path = strings.Replace(path, "{"+pathParams[i]+"}", arg, -1)
 			}
-			newErr.model = v
-
-			ParseResponse(localVarHttpResponse, &newErr)
 
-			return localVarReturnValue, localVarHttpResponse, newErr
-		}
-
-		ParseResponse(localVarHttpResponse, &newErr)
-
-		return localVarReturnValue, localVarHttpResponse, newErr
+			return path
+		},
+		StatusCodeResultsFactoryMap: map[int]func() interface{}{
+			200: func() interface{} { return &Response{} },
```

Comment:
Could you show how the case `200` is used? I can only find the usage inside `handleErrorResponse` function. If it is really not used, let's remove it to avoid confusion.

Reply:
Same as above.